### PR TITLE
feat(sentry): add starlite sentry integration

### LINKED ISSUE #52: bug(sentry): the sentry asgi middleware doesn't really work for us
Need to get the sentry integration task done for starlite org and use that instead.

### PR DESCRIPTION
**Pull request by gazorby**
Closes #52

### CODE REVIEW

File: src/starlite_saqlalchemy/sentry.py [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
+
+    class SamplingContext(TypedDict):
+        """Sentry context sent to traces sampler function."""
+
+        asgi_scope: HTTPScope
+        parent_sampled: bool | None
+        transaction_context: dict[str, Any]
+
+
+def sentry_traces_sampler(sampling_context: Mapping[str, Any]) -> float:
```
Comment (peterschutt) [inline src/starlite_saqlalchemy/sentry.py:26]: ahh this is cool:)

**Comment by peterschutt** [comment]
So the way the sentry integration globally patches starlite means that if you run that failing test on its own, it passes as  `sentry.configure()` has never been called. Once it is called at all, that test will fail as there is an exc. handler added in the init wrapper (as you know!).

So I think we just disable sentry for the tests, thoughts @gazorby?

**Comment by gazorby** [comment]
Yes it's better to disable it for unit tests, but maybe we can do something in the integration environement? I just saw that `LifespanManager` trigger lifecycle hooks, so we could try something like that:

```python
from starlite import Starlite
from sentry_sdk.integrations.starlite import SentryStarliteASGIMiddleware

def test_sentry(app):
    old_init = Starlite.__init__
    async with LifespanManager(app):
        assert isinstance(app.middleware[0], SentryStarliteASGIMiddleware)
    Starlite.__init__ = old_init
```

**Comment by gazorby** [comment]
Also, what do you think about disabling sentry when ENVIRONMENT == "local"?

**Comment by peterschutt** [comment]
> but maybe we can do something in the integration environement?

It would be nice to have the sentry integration enabled somewhere within the tests to pick up any side-effects of having it activated before deploying. I'd even be happy to run the whole integration test suite with it activated - perhaps we can use the `app` fixture in `tests/integration/conftest.py`, make it a `yield` fixture, and have the logic to "un-patch" starlite after the yield.

I've still got to fix that integration test that has been x-failed.

If you want to do it in this PR or another, I'm easy.

> Also, what do you think about disabling sentry when ENVIRONMENT == "local"?
Yes, makes sense, and have the implicit setting able to be overridden by environment like we did for reloading.

**Pull request merged**